provider/pkg/provider/remote/connection.go

```diff
@@ -107,16 +117,73 @@ func (con *Connection) SShConfig() (*ssh.ClientConfig, error) {
 	return config, nil
 }
 
+func (con *Connection) ProxySShConfig() (*ssh.ClientConfig, error) {
+	config := &ssh.ClientConfig{
+		User:            *con.ProxyUser,
```

Comment:
The tests at https://github.com/pulumi/pulumi-command/actions/runs/4527001624/jobs/7972532321#step:25:321 appear to be triggering a panic here.

Looks like we aren't handling the cases where there is no proxy configuration being provided.